### PR TITLE
Updating MAVLink camera/video commands.

### PR DESCRIPTION
**Pull request by dogmaphobic**
@DonLakeFlyer The thing to look for are the `CameraSection::_scanXxx` methods. They used to compare all 7 parameters in some cases. I reduced to only compare those you actually set, skipping the *reserved* ones.

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
> They used to compare all 7 parameters in some cases. I reduced to only compare those you actually set, skipping the reserved ones.

Yeah, I was trying to be pretty anal about sucking those commands back up. But I reality I think this is perfectly fine. Thanks.

**Comment by DonLakeFlyer** [comment]
Did you look at the the CameraSection unit tests. They may still pass, but they may need to be updated where they create mission items for each camera command type to put NaNs in the right places for completeness.

**Comment by dogmaphobic** [comment]
I went through with a fine comb over every instance of those messages throughout the code, including through `CameraSectionTest.cc` but it would not hurt for you to do a quick check.

File: src/MissionManager/CameraSectionTest.cc [hunk 1]
```diff
@@ -90,20 +88,19 @@ void CameraSectionTest::init(void)
...
                                                             MAV_CMD_IMAGE_START_CAPTURE,
                                                             MAV_FRAME_MISSION,
-                                                            0,                               // camera id = 0, all cameras
+                                                            0,                              // camera id = 0, all cameras
                                                             0,                              // Interval (none)
                                                             1,                              // Take 1 photo
-                                                            -1,                             // Max horizontal resolution
-                                                            -1,                             // Max vertical resolution
-                                                            0, 0,                           // param 6-7 not used
+                                                            NAN, NAN, NAN, NAN,             // param 4-7 not reserved
```
Comment (dogmaphobic) [inline src/MissionManager/CameraSectionTest.cc]: Oops... `"not" reserved` ain't exactly right but I won't restart a build just for that :P

**Comment by DonLakeFlyer** [comment]
Sorry somehow I missed the CameraSectionTest changes. I hate the way GitHub shows diffs. Also looked through again. The unit tests for CameraSection are pretty anal since I had so many problems with it. So if they pass it should be good.

**Comment by dogmaphobic** [comment]
@DonLakeFlyer How can I tell what test failed and why?

**Comment by DonLakeFlyer** [comment]
It's all in the log file for the run: https://travis-ci.org/mavlink/qgroundcontrol/jobs/247354153 which you can get to from travis. Looks like the scanning is broken. In looking at it, it's because the camera section unit tests has testing for writing non-zero values into reserved params. Previously that would fail the scan, but now since those param tests are removed in the scan they now succeed even when nonzero values are in there.

**Comment by DonLakeFlyer** [comment]
So CameraSectionTest needs to be updated to only write bad values into non-reservered params for each command.

**Comment by dogmaphobic** [comment]
I think I found it...

**Comment by dogmaphobic** [comment]
I have no clue how those unit tests work. I found the tests that were causing errors but I do not know why so as I could not "fix" them, I commented them off (instead of removing as you may want to either fix, change or remove them yourself).

**Pull request merged**